### DISCUSSION TITLE
Unable to bypass zones on Partition 2.

### DISCUSSION THREAD

**Discussion by 949BFN** [ufodone/envisalink_new#118] [Q&A] [answered]
My DCS Panel has 2 partitions. All functions work fine for both partitions (i.e. Arming, Disarming etc.) However, I'm unable to bypass Zones on the second partition. (Bypassing a zone on the first partition works fine.)
My partition/zone configuration looks like this:
For Partitions: 1,2
For Zones: 1,2,3,4,5,6,12,15,17,18,
The zones are not in sequence because zones 1, 2, 3, 4, 5, 6, 12, 15, 17, 18 are on partition 1 and zones 9, 10, 11, 13, 14, 23 are on partition 2. I thought listing the zones according to the sequence on the two partitions might help with my issue.

Additional info:
I discovered I could not bypass zones on the second partition when a beam started giving me trouble by staying in the "open" state. I wanted to bypass the faulty zone to arm the second partition. 
The Partition 2 Keypad sensor shows "Zone/Sensor Fault" in Home Assistant.
Don't know if it is important, but I'm able to bypass zones on the second partition with the Eyezon app. Once I've bypassed the zone with the Eyzeon app, I can arm that zone in Home Assistant.
Thank you.

**Comment by ufodone** [comment, marked as answer]
The integration currently sends the zone bypass keystrokes to partition 1 which likely explains your issue.  At the moment it does not have any awareness of the zone to partition mapping so it doesn't know where to send the command.

Feel free to open an [issue](https://github.com/ufodone/envisalink_new/issues) to track this.  It likely would be a bit of a bigger change though due to the lack of zone/partition mapping so it might take a bit of time for me to get to.

While not ideal, in the meantime you might be able to hack something up using a combination of the `alarm_keypress` service call and maybe a template switch.  E.g. the below should toggle the bypass on zone 9, partition 2.

```
service: envisalink_new.alarm_keypress
target:
  entity_id: alarm_control_panel.partition_2
data:
  keypress: "*209#"
```

**Reply by 949BFN** [reply]
@ufodone 
I believe the latest update has  resolved this issue. (Thank you for that.) I, however, can't figure out how or where to configure to deal with a two partition system. 
The instructions tell us "The zone and partition list accepts a comma separated list of numerical zones/partitions as well as ranges. For example: 1-2,4-8,16-18,20-29"
I don't understand how the partition information can be added.
I guess my question is, how do I link specifice zones to specific partitions.
Thank you.

**Reply by ufodone** [reply]
Under `Configuration->Partition Assignments` you should be presented with a text entry box for each partition you  have configured and so you need to fill in the zones assigned to each partition in the appropriate field.  For example if you have two partitions configured (in the `Basic` settings), you could do something like:

<img width="396" alt="Screenshot 2025-01-30 at 10 56 00 AM" src="https://github.com/user-attachments/assets/96fd6dd6-3b46-46d6-9a37-e0bacbfd2c07" />

Which assigns zones 1, 2, 3, 4, 7, and 8 to partition 1 and zones 5, 6, 9, 10, 11, 12, 13, 14, 15, 16 to partition 2.

**Reply by 949BFN** [reply]
Much appreciated!

**Comment by 949BFN** [comment]
Thank you, that is much appreciated. 
By the way, the code *209# did not work for me. *109# (with partition 2 selected as the entity) does the trick. 
This gave me an idea—I believe there is a code to force-arm the partition even if there is a zone fault. I'm going to look for that and see if I can't use that code in the automation to arm the partition. That way, I won't have to worry that the alarm fails to arm because of some fault or a door is not properly closed.
